### PR TITLE
fixes for Julia 0.6, dropping support for 0.4

### PR DESCRIPTION
**Pull request by dehann**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,3 @@
-julia 0.4
-DataStructures 0.3.8
-Compat 0.7.16
+julia 0.5
+DataStructures 0.6.0
```
Comment (tkelman) [inline REQUIRE]: wouldn't always using PriorityQueue from 0.5 or higher be good enough?

Comment (dehann) [inline REQUIRE]: Hi, PriorityQueue moved to DataStructures in 0.6 right -- not sure I understand what you mean?

Comment (dehann) [inline REQUIRE]: Ah you mean in drop the minimum REQUIRE for DataStructures

**Comment by dehann** [comment]
see #240

**Comment by tkelman** [comment]
LGTM. Note that `const A = B` doesn't need `@compat` unless the left hand side has a type parameter, and `take!` doesn't need `@compat` either since it's a normal method rather than a syntax rewrite. But it doesn't hurt to call `@compat` even when unnecessary.

**Comment by dehann** [comment]
Thanks, will do more clean-up on later.

And thanks @maxvoxel8

**Comment by dpo** [comment]
New release? Currently, Metis.jl is broken because of the bug fixed in this PR.

**Comment by dehann** [comment]
Hi @dpo , will do -- maybe just hang tight for a few days if that is okay please?

**Comment by dehann** [comment]
Hi, finally tagged new version in METADATA at JuliaLang/METADATA.jl#11202

**Pull request merged**